benches/Kinetic.Benchmarks/Benchmarks.cs

```diff
@@ -52,7 +53,16 @@ private void Set<T>(ref T field, T value, [CallerMemberName] string property = "
     protected class KineticTestObject : ObservableObject
     {
         private int _field;
-        public Property<int> Property => base.Property(ref _field);
+        private int _fieldWithHook;
+
+        public Property<int> Property =>
+            base.Property(ref _field);
+
+        public Property<int> PropertyWithHook =>
+            base.Property(ref _fieldWithHook);
+
+        public KineticTestObject() =>
+            Preview<int>(PropertyWithHook, static changing => changing.Select(value => value));
```

Comment:
Unfortunately, it comes from the implicit cast operator to `T` in the property types. Perhaps, the operator should be removed, but then code will be more verbose.

Another option is accepting a field reference, but that doesn't sound as it works only with the corresponding property change and not with direct field access.